src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -117,6 +125,13 @@ public Set<String> getTargetMethods() {
 
   @Override
   public Visitable visit(ClassOrInterfaceDeclaration decl, Void p) {
+    if (decl.getExtendedTypes().isNonEmpty()) {
+      // note that since Specimin does not have access to the class paths of the project, all the
+      // unsolved methods related to inheritance will be placed in the parent class, even if there
+      // is a grandparent class and so forth.
+      parentClass = decl.getExtendedTypes().get(0).resolve().getQualifiedName();
```

Comment:
I haven't thought of this case. If there are inner classes, it will definitely break the algorithm. Thank you for pointing out. I will try change parentClass to a map rather than a String to see if it helps.